app/Console/Commands/ForceDeletePosts.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Rogue\Console\Commands;
+
+use Rogue\Models\Post;
+use Illuminate\Console\Command;
+use Rogue\Managers\PostManager;
+
+class ForceDeletePosts extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'rogue:forceDeletePosts
+                            {type}
+                            {source}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Force deletes posts in Rogue by type and source';
+
+    /**
+     * The post manager.
+     * @var PostManager
+     */
+    protected $postManager;
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct(PostManager $postManager)
+    {
+        // Post Manager Instance.
+        $this->postManager = $postManager;
+
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        if (! $this->confirm('Are you sure you want to delete posts of type ' . $this->argument('type') . 'from ' . $this->argument('source'))) {
+            return;
+        }
+
+        info('rogue:forceDeletePosts: Starting to force delete posts by type and source.');
+
+        $posts = Post::where('type', $this->argument('type'))->where('source', $this->argument('source'))->get();
+
+        if ($posts->isNotEmpty()) {
+            $posts->map(function ($post, $key) {
```

Comment:
Let's swap this to use `forEach` like @katiecrane suggested!

Reply:
ok! @DFurnes just curious - why is it better to use a `forEach` statement here instead of `map`?

Reply:
This should use [`forEach`](https://laravel.com/docs/5.6/collections#method-forEach) since we're not returning anything. You'd use [`map`](https://laravel.com/docs/5.6/collections#method-map) to iterate over a collection & modify each element (and so since we're not returning anything, this line technically makes and returns a new empty collection).

Reply:
ah I see - good to know, thanks! Updated :)